livedata/src/main/java/io/getstream/chat/android/livedata/ChatDomainImpl.kt

```diff
@@ -467,28 +468,12 @@ class ChatDomainImpl private constructor(
         return activeQueryMapImpl[queryChannelsEntity]!!
     }
 
-    suspend fun queryEvents(cids: List<String>): List<ChatEvent> {
-        val limit = 100
-        var offset = 0
-        val maxEvents = 500
-        val allEvents = mutableListOf<ChatEvent>()
-
-        while (true) {
-            val call = client.replayEvents(cids, syncState?.lastSyncedAt, limit, offset)
-            val response = call.execute()
-            if (response.isError) {
-                // TODO: what is the best error type for this?
-                throw Error(response.error())
-            }
-            val events = response.data()
-            allEvents.addAll(events)
-            if (events.size <= limit || offset >= maxEvents) {
-                break
-            }
-            offset += limit
+    private fun queryEvents(cids: List<String>): List<ChatEvent> {
+        val response = client.getSyncHistory(cids, syncState?.lastSyncedAt ?: NEVER).execute()
+        if (response.isError) {
+            throw response.error()
```

Comment:
Instead of throw an exception, is not possible to return an error?

Reply:
dont we need pagination on this? depends on how many items we can fetch at once of course. good to check with the backend team.

Reply:
The current implementation of sync endpoint client in LLC doesn't support any pagination. I'll check it with the backend team when working on the sync itself.